### PR TITLE
Warn about using loose assembly.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/1824 https://github.com/ethereum/solidity/issues/1821 https://github.com/ethereum/solidity/issues/1822 https://github.com/ethereum/solidity/issues/1825

Depends on https://github.com/ethereum/solidity/pull/3476

### CODE REVIEW

**Comment by chriseth** [comment]
Ready for review.

**Comment by axic** [comment]
The assembly part looks good. Need to finish reviewing the scoping changes in the other PR first.

**Comment by chriseth** [comment]
Rebased.

File: libsolidity/inlineasm/AsmAnalysis.cpp [hunk 1]
```diff
@@ -54,15 +54,21 @@ bool AsmAnalyzer::analyze(Block const& _block)
 
 bool AsmAnalyzer::operator()(Label const& _label)
 {
-	solAssert(m_flavour == AsmFlavour::Loose, "");
+	checkLooseFeature(
+		_label.location,
+		"The use of labels is deprecated. Please use if, switch, for or function calls instead."
```
Comment (axic) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: Should quote `if`, `switch` and `for`

**Comment by chriseth** [comment]
@axic good to merge?

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Fails on:
```
inline_assembly_050_bare_instructions - SolidityNameAndTypeResolution
inline_assembly_050_jump - SolidityNameAndTypeResolution
inline_assembly_050_labels - SolidityNameAndTypeResolution
inline_assembly_050_leave_items_on_stack - SolidityNameAndTypeResolution
inline_assembly_050_literals_on_stack - SolidityNameAndTypeResolution
inline_assembly_storage_variable_access_out_of_functions - SolidityNameAndTypeResolution
```

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 2]
```diff
@@ -6112,6 +6122,187 @@ BOOST_AUTO_TEST_CASE(inline_assembly_calldata_variables)
...
+					address
+					pop
+				}
+			}
+		}
+	)";
+	CHECK_ALLOW_MULTI(text, (std::vector<std::pair<Error::Type, std::string>>{
+		{Error::Type::SyntaxError, "The use of non-functional"},
+		{Error::Type::SyntaxError, "The use of non-functional"},
+		{Error::Type::Warning, "Experimental features are turned on"}
```
Comment (axic) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp]: Fails because of these.

**Comment by chriseth** [comment]
Should work now. Will rebase.

**Review by axic — APPROVED** [review]

**Pull request merged**